tracer/src/Datadog.Trace/Baggage.cs

```diff
@@ -0,0 +1,354 @@
+﻿// <copyright file="Baggage.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+
+#nullable enable
+
+namespace Datadog.Trace;
+
+/// <summary>
+/// Baggage is a collection of name-value pairs that are propagated to downstream services.
+/// </summary>
+internal class Baggage : IDictionary<string, string>
+{
+    private static readonly AsyncLocal<Baggage> AsyncStorage = new();
+
+    private static readonly List<KeyValuePair<string, string>> EmptyList = [];
+
+    private List<KeyValuePair<string, string>>? _items;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Baggage"/> class.
+    /// </summary>
+    public Baggage()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Baggage"/> class using the specified items.
+    /// </summary>
+    /// <param name="items">The baggage items.</param>
+    public Baggage(IEnumerable<KeyValuePair<string, string>> items)
+    {
+        if (items != null!)
+        {
+            _items = [..items];
```

Comment:
I forgot about this `..` syntax very nice

Reply:
Thank Rider's suggestions 😅